### PR TITLE
fix: change audit descriptions to indicate unsuccessful attempts

### LINKED ISSUE #11607: Failed attempts to delete users show up in audit log as successful
I attempted to delete a user that had a stopped workspace and got the expected "You cannot delete a user that has workspaces. Delete their workspaces and try again!" error.

When looking in the audit log, I noticed that there was an audit entry that suggested that I successfully deleted the user:
```
<my-username> has deleted user <target-username>
```
Since my deletion attempt aborted due to error, I'd instead expect the entry to either be absent or something like:
```
<my-username> unsuccessfully attempted to delete user <target-username>
```

### PR DESCRIPTION
**Pull request by spikecurtis**
fixes #11607

### CODE REVIEW

**Comment by spikecurtis** [comment]
* **#13897** <a href="https://app.graphite.dev/github/pr/coder/coder/13897?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @spikecurtis and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: coderd/audit_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/database"
+)
+
+func TestAuditLogDescription(t *testing.T) {
+	t.Parallel()
+	// nolint: paralleltest // no longer need to reinitialize loop vars in go 1.22
+	for _, tc := range []struct {
```
Comment (aslilac) [inline coderd/audit_internal_test.go:14]: could we move this out to a variable? generally not a huge fan of large inline variables like this

**Review by aslilac — APPROVED** [review]
I wrote a whole thing about how I'm not crazy about ignoring all of the errors in here, but then I went to look up `WriteString` to see what errors it could return (I assumed allocation errors) only to see that the docs guarantee that it will only ever return `nil`. I find this frustrating, because our linting that requires the `_, _ =` makes it look like you're tossing tons of errors, but the docs explicitly say that you're not if you look them up.

maybe we could leave a comment that it'll never _actually_ return an error despite the return type, to avoid confusion in the future. not a huge deal, I just wanted to flag the emotional journey this just sent me on. 😅

**Comment by spikecurtis** [comment]
I appreciate the detailed review that you took the time to check `WriteString` docs. Sorry for winding you up about error checking!

**Comment by spikecurtis** [comment]
### Merge activity

* **Jul 16, 2:47 PM EDT**: @spikecurtis merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coder/coder/13897).

**Pull request merged**